odl/discr/grid.py

```diff
@@ -1195,8 +1203,12 @@ def uniform_sampling_fromintv(intv_prod, num_nodes):
     >>> grid = uniform_sampling_fromintv(rbox, (3, 3))
     >>> grid.coord_vectors
     (array([-1.5, -1. , -0.5]), array([ 2. ,  2.5,  3. ]))
+
+    See also
+    --------
+    uniform_sampling : Sample an implicitly created `IntervalProd`
     """
-    num_nodes = np.atleast_1d(num_nodes).astype('int64')
+    num_nodes = np.atleast_1d(num_nodes).astype('int64', casting='safe')
```

Comment:
But AFAIK it's the first step on the harshness ladder. Compare:

- `same_kind` -> only from integer types, same result
- `equiv` -> only byte order changes, excludes even most integer types
- `no` -> self-explanatory
- `unsafe` -> old version, any cast
- `safe` -> current

I changed it because you thought it would be better to not silently cast, which I think, too.